smack-core/src/main/java/org/jivesoftware/smack/altConnections/HttpLookupMethod.java

```diff
@@ -0,0 +1,187 @@
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> parseXrdLinkReferencesFor(XmlPullParser parser, String relation) throws IOException, XmlPullParserException, URISyntaxException {
+        final String xrd_namespace = "http://docs.oasis-open.org/ns/xri/xrd-1.0";
+
+        List<URI> uriList = new ArrayList<>();
+        int initialDepth = parser.getDepth();
+
+        XmlPullParser.TagEvent tag = parser.nextTag();
+        loop: while (true) {
+            switch (tag) {
+            case START_ELEMENT:
+                String name = parser.getName();
+                String namespace = parser.getNamespace();
+                String rel = parser.getAttributeValue("rel");
+
+                if (!namespace.equals(xrd_namespace) || !name.equals("Link") || !rel.equals(relation)) {
+                    tag = parser.nextTag();
+                    continue loop;
+                }
+                String endpointUri = parser.getAttributeValue("href");
+                uriList.add(new URI(endpointUri));
+                tag = parser.nextTag();
+                break;
+            case END_ELEMENT:
+                if (parser.getDepth() == initialDepth) {
+                    break loop;
+                }
+                tag = parser.nextTag();
+                break;
+            }
+        }
+        return uriList;
+    }
+
+    /**
+     * Get remote endpoints for the provided LinkRelation from provided XmlPullParser.
+     *
+     * @param parser XmlPullParser that contains LinkRelations
+     * @param relation type of endpoints specified by the given LinkRelation
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> parseXrdLinkReferencesFor(XmlPullParser parser, LinkRelation relation) throws IOException, XmlPullParserException, URISyntaxException {
+        return parseXrdLinkReferencesFor(parser, relation.attribute);
+    }
+
+    /**
+     * Get remote endpoints for the given LinkRelation from host.
+     *
+     * @param xmppServiceAddress address of host
+     * @param relation LinkRelation as a string specifying type of connection
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> lookup(DomainBareJid xmppServiceAddress, String relation) throws IOException, XmlPullParserException, URISyntaxException {
```

Comment:
Please move this method and the method below to the top of the class, since those are the most important high level methods.